flang/lib/Lower/ConvertCall.cpp

```diff
@@ -2436,8 +2436,9 @@ genCustomIntrinsicRef(const Fortran::evaluate::SpecificIntrinsic *intrinsic,
                                          getActualFortranElementType());
       break;
     case fir::LowerIntrinsicArgAs::Inquired:
-      TODO(loc, "Inquired non-optional arg to intrinsic with custom handling");
-      return;
+      exv = Fortran::lower::convertToBox(loc, converter, actual, stmtCtx,
```

Comment:
I do not think this is the right function to call, convertToBox will generate the dereference of the pointer (the fir.load), while inquired arguments should not be touched until inquired (they may be absent, deallocated....).

I would expect `Fortran::lower::translateToExtendedValue(loc, builder, actual, stmtCtx);` here.

Reply:
That makes lots of sense! I'll update that.

Reply:
Done